instance/github.py

```diff
@@ -149,26 +140,48 @@ class PR:
     """
     Representation of a GitHub Pull Request
     """
-    # pylint: disable=too-many-arguments
-    def __init__(self, number, source_fork_name, target_fork_name, branch_name, title, username, body=''):
+
+    def __init__(self, target_fork_name, number):
         self.number = number
-        self.fork_name = source_fork_name
         self.repo_name = target_fork_name
-        self.branch_name = branch_name
-        self.title = title
-        self.username = username
-        self.body = body
+        self.update_data()
+
+    def update_data(self):
+        self._data = get_pr_by_number(target_fork_name, number)
+        self._extra_settings = get_settings_from_pr_body(self.body)
+        self._extra_settings_dict = yaml.load(self.extra_settings) or {}
```

Comment:
This is a performance optimization, yes.  Once we start accessing more properties parsed from the PR body, we would otherwise have to reparse the whole body, extract the YAML, parse it and extract whatever we need every single time we want to access one of these properties, which seems excessive.

The point of the refactoring is that this way there is practically no chance of cached data getting out of sync, since the attributes are only accessed in `update_data()`, where all of them are updated at once.

Since we don't seem to get an agreement on this, I'll create a new PR against the master branch with a stripped down version on the second commit in this PR.